### DISCUSSION TITLE
Cannot push changes to GitHub

### DISCUSSION THREAD

**Discussion by MarcoPozuv** [community/community#85677] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hello,

I did some modifications to my local system and I want to push those changes to a branch in my GitHub repository. I use the command "git push --set-upstream origin <name of the branch>". In the terminal, it asks for my username and password. I put my username and then when I tried to write the password the terminal did not show any letter, even though, I entered my password as normal, clicked enter and this message showed up

remote: Support for password authentication was removed on August 13, 2021.
remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
fatal: Authentication failed

Do you have any idea how to solve this?
Thanks

**Comment by bonizario** [comment]
The terminal intentionally does not display the password or placeholder characters like asterisks (*) for security reasons. This is a common practice in Unix-based systems to prevent shoulder surfing attacks.

So, even though you don't see the password, it is being entered. After typing your password, press Enter. If the password is correct, the command will execute successfully.

**Reply by MarcoPozuv** [reply]
Thanks for the reply

But actually that is the problem, after typing my password and clicking enter, the error message still appears
I've verified my password and username a lot of times and they are ok

**Comment by LuksJobs** [comment, marked as answer]
The error you're encountering is because GitHub has removed support for password authentication over HTTPS, and it's recommending the use of token-based authentication. To resolve this issue, you need to generate a personal access token (PAT) and use it as your password when pushing changes.

#### Here's how you can do it:

#### Generate a Personal Access Token:

1. Go to your GitHub account settings.
2. Navigate to "`Developer settings`" > "`Personal access tokens`" > "`Generate token`" (or click here).
3. Provide a meaningful name for the token, select the scopes or permissions your token needs, and click "Generate token."
4. Copy the generated token to your clipboard.

#### Update Git Credentials:

On your local machine, open a terminal.

#### Run the following command to update the credentials for your Git configuration:

`git credential reject`

#### Then try pushing again:

`git push --set-upstream origin <your-branch-name>`

It will prompt you for a username; provide your GitHub username.

When prompted for the password, paste the personal access token you generated earlier.

Update Git Remote URL (Optional):

If you want to avoid entering your username and token each time you push, you can update the remote URL to include the token.

#### Run the following command:

`git remote set-url origin https://username:token@github.com/username/repository.git`

Replace username with your GitHub username, token with the personal access token, and repository with your repository name.

After performing these steps, your local Git configuration should be updated, and you should be able to push changes to the repository using your personal access token.

Remember to keep your personal access token secure, as it provides access to your repositories. If at any point you feel your token has been compromised, you can regenerate it in your GitHub settings.

**Reply by MarcoPozuv** [reply]
Thank you. The token and this command solved it: https://<token>@github.com/<username>/<Repository name>
